src/backend/postmaster/postmaster.c

```
@@ -998,6 +995,33 @@ PostmasterMain(int argc, char *argv[])
 	if (!SelectConfigFiles(userDoption, progname))
 		ExitPostmaster(2);
 
+	/* Validate gp_dbid and gp_contentid */
+	if (GpIdentity.dbid < 0 || GpIdentity.segindex < -1)
+	{
+		if (GpIdentity.dbid != UNINITIALIZED_GP_IDENTITY_VALUE ||
+			GpIdentity.segindex != UNINITIALIZED_GP_IDENTITY_VALUE)
+			ereport(FATAL, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							errmsg("Invalid gp_dbid(%d) or gp_contentid(%d)",
+									GpIdentity.dbid, GpIdentity.segindex),
+							errhint("Set both gp_dbid and gp_contentid properly, "
+									"or leave them both uninitialized")));
+
+		if (Gp_role != GP_ROLE_UTILITY)
```

If the command line contains `-c gp_role=execute` then I feel we should error out, instead of emitting a log and overriding the command line.  It's better to force correct options be used on a command line than to ignore them.

---

Reasonable.